src/base/torrentfileswatcher.cpp

```diff
@@ -375,56 +364,60 @@ void TorrentFilesWatcher::store() const
     QJsonObject jsonObj;
     for (auto it = m_watchedFolders.cbegin(); it != m_watchedFolders.cend(); ++it)
     {
-        const QString &watchedFolder = it.key();
+        const Path &watchedFolder = it.key();
         const WatchedFolderOptions &options = it.value();
-        jsonObj[watchedFolder] = serializeWatchedFolderOptions(options);
+        jsonObj[watchedFolder.toString()] = serializeWatchedFolderOptions(options);
```

Comment:
> My conclusion so far is that only when passing value to Qt functions or qbt internal functions can use toString(), otherwise it is toNative()

It's quite the opposite. It makes sense to convert paths to a native form only when it is really necessary, i.e. when we call third-party functions or show them to the user. Everything else is extra heat generation (don't forget that they are stored in uniform format).

Note, that I try to avoid changing format in WebAPI etc.(although they are used inconsistently there).